hedera-node/src/main/java/com/hedera/services/ServicesState.java

```diff
@@ -339,159 +227,176 @@ public void noMoreTransactions() {
-		return uniqueTreasuryOwnedAssociations;
+		return metadata.getUniqueTreasuryOwnershipAssociations();
 	}
 
-	void setUniqueTokenAssociations(FCOneToManyRelation<Integer, Long> uniqueTokenAssociations) {
-		this.uniqueTokenAssociations = uniqueTokenAssociations;
+	private void internalInit(Platform platform, BootstrapProperties bootstrapProps) {
+		setImmutable(false);
+
+		networkCtx().setStateVersion(StateVersions.CURRENT_VERSION);
+		diskFs().checkHashesAgainstDiskContents();
+
+		final var selfId = platform.getSelfId();
+		ServicesContext ctx;
+		if (CONTEXTS.isInitialized(selfId.getId())) {
+			ctx = CONTEXTS.lookup(selfId.getId());
+		} else {
+			final var properties = new StandardizedPropertySources(bootstrapProps);
+			ctx = new ServicesContext(selfId, platform, this, properties);
+		}
+		metadata = new StateMetadata(ctx);
+
+		contextInitializer.accept(this, ctx);
+		CONTEXTS.store(ctx);
+		logSummary();
+
+		log.info("  --> Context initialized accordingly on Services node {}", selfId);
 	}
 
-	void setUniqueOwnershipAssociations(FCOneToManyRelation<Integer, Long> uniqueOwnershipAssociations) {
-		this.uniqueOwnershipAssociations = uniqueOwnershipAssociations;
+	private void createGenesisChildren(AddressBook addressBook, long seqStart) {
+		setChild(StateChildIndices.UNIQUE_TOKENS, new FCMap<>());
+		setChild(StateChildIndices.TOKEN_ASSOCIATIONS, new FCMap<>());
+		setChild(StateChildIndices.TOPICS, new FCMap<>());
+		setChild(StateChildIndices.STORAGE, new FCMap<>());
+		setChild(StateChildIndices.ACCOUNTS, new FCMap<>());
+		setChild(StateChildIndices.TOKENS, new FCMap<>());
+		setChild(StateChildIndices.NETWORK_CTX, genesisNetworkCtxWith(seqStart));
+		setChild(StateChildIndices.DISK_FS, new MerkleDiskFs());
+		setChild(StateChildIndices.SCHEDULE_TXS, new FCMap<>());
+		setChild(StateChildIndices.RECORD_STREAM_RUNNING_HASH, genesisRunningHashLeaf());
+		setChild(StateChildIndices.ADDRESS_BOOK, addressBook);
 	}
 
-	void setUniqueTreasuryOwnershipAssociations(FCOneToManyRelation<Integer, Long> uniqueTreasuryOwnershipAssociations) {
-		this.uniqueTreasuryOwnedAssociations = uniqueTreasuryOwnershipAssociations;
+	private RecordsRunningHashLeaf genesisRunningHashLeaf() {
+		final var genesisRunningHash = new RunningHash();
+		genesisRunningHash.setHash(EMPTY_HASH);
+		return new RecordsRunningHashLeaf(genesisRunningHash);
 	}
 
-	private void releaseFcotmrIfNonNull() {
-		if (uniqueTokenAssociations != null) {
-			uniqueTokenAssociations.release();
-		}
-		if (uniqueOwnershipAssociations != null) {
-			uniqueOwnershipAssociations.release();
-		}
-		if (uniqueTreasuryOwnedAssociations != null) {
-			uniqueTreasuryOwnedAssociations.release();
-		}
+	private MerkleNetworkContext genesisNetworkCtxWith(long seqStart) {
+		return new MerkleNetworkContext(
+				UNKNOWN_CONSENSUS_TIME,
+				new SequenceNumber(seqStart),
+				seqStart - 1,
+				new ExchangeRates());
+	}
+
+	/* --- Only used by unit tests --- */
```

Comment:
Minor suggestion, feel free to ignore if you don't like it. 

You could clean up this class by removing these test-only methods. Create a new class `AccessibleServicesState` that extends `ServicesState`. Add the test-only methods there. If the test methods need to access member variables, make them protected in the base class.

In the unit test, instantiate an `AccessibleServicesState` and test on that.

Not sure how you are unit testing, so this may or may not make sense in that context.

Reply:
Very clever! If not for time constraints, would definitely do. :/